external/bugswarm/scripts/download_dataset.py

```
@@ -0,0 +1,282 @@
+    """Return failure log URL/path for a job via API get_build_log, or canonical BugSwarm log URL."""
+    if api is None:
+        return ""
+    job_id = (failed_job_id or "").strip()
+    if not job_id:
+        return ""
+    try:
+        resp = api.get_build_log(job_id)
+        if getattr(resp, "ok", False) and getattr(resp, "url", None):
+            return resp.url
+        # Fallback: canonical BugSwarm log URL
+        return f"https://www.bugswarm.org/artifact-logs/{job_id}/"
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)